tests/testthat/test-excel_time_to_numeric.R

```diff
@@ -0,0 +1,161 @@
+  )
+  expect_equal(
+    excel_time_to_numeric("8:00am"),
+    excel_time_to_numeric("8:00Am")
+  )
+  expect_equal(
+    excel_time_to_numeric("8:00am"),
+    excel_time_to_numeric("8:00aM")
+  )
+  expect_equal(
+    excel_time_to_numeric("8:00am"),
+    excel_time_to_numeric("8:00 AM")
+  )
+})
+
+test_that("excel_time_to_numeric, 24-hour times work (zero-padded hours or not)", {
+  expect_equal(excel_time_to_numeric("8:00"), 8 * 3600)
+  expect_equal(excel_time_to_numeric("08:00"), 8 * 3600)
+  expect_equal(excel_time_to_numeric("08:10"), 8 * 3600 + 10 * 60)
+  expect_equal(excel_time_to_numeric("8:10:05"), 8 * 3600 + 10 * 60 + 5)
+  expect_equal(excel_time_to_numeric("21:05"), 21 * 3600 + 5 * 60)
+  expect_equal(excel_time_to_numeric("21:05:20"), 21 * 3600 + 5 * 60 + 20)
+})
+
+test_that("excel_time_to_numeric, 24-hour times work (zero-padded hours or not)", {
+  expect_equal(excel_time_to_numeric("8:00"), 8 * 3600)
+  expect_equal(excel_time_to_numeric("08:00"), 8 * 3600)
+  expect_equal(excel_time_to_numeric("08:10"), 8 * 3600 + 10 * 60)
+  expect_equal(excel_time_to_numeric("8:10:05"), 8 * 3600 + 10 * 60 + 5)
+  expect_equal(excel_time_to_numeric("21:05"), 21 * 3600 + 5 * 60)
+  expect_equal(excel_time_to_numeric("0:05"), 5 * 60)
+  expect_equal(excel_time_to_numeric("00:05"), 5 * 60)
+  expect_equal(excel_time_to_numeric("21:05:20"), 21 * 3600 + 5 * 60 + 20)
+})
+
+test_that("excel_time_to_numeric, POSIX times on 1899-12-31 work", {
+  expect_equal(excel_time_to_numeric("1899-12-31 8:00"), 8 * 3600)
+  expect_equal(excel_time_to_numeric("1899-12-31 08:00"), 8 * 3600)
+  expect_equal(excel_time_to_numeric("1899-12-31 08:10"), 8 * 3600 + 10 * 60)
+  expect_equal(excel_time_to_numeric("1899-12-31 8:10:05"), 8 * 3600 + 10 * 60 + 5)
+  expect_equal(excel_time_to_numeric("1899-12-31 21:05"), 21 * 3600 + 5 * 60)
+  expect_equal(excel_time_to_numeric("1899-12-31 0:05"), 5 * 60)
+  expect_equal(excel_time_to_numeric("1899-12-31 00:05"), 5 * 60)
+  expect_equal(excel_time_to_numeric("1899-12-31 21:05:20"), 21 * 3600 + 5 * 60 + 20)
+})
+
+test_that("excel_time_to_numeric, POSIX times ignore extra text (which is hopefully a time zone)", {
+  expect_equal(excel_time_to_numeric("1899-12-31 8:00 foo"), 8 * 3600)
+  expect_equal(excel_time_to_numeric("1899-12-31 08:00 foo"), 8 * 3600)
+  expect_equal(excel_time_to_numeric("1899-12-31 08:10 foo"), 8 * 3600 + 10 * 60)
+  expect_equal(excel_time_to_numeric("1899-12-31 8:10:05 foo"), 8 * 3600 + 10 * 60 + 5)
+  expect_equal(excel_time_to_numeric("1899-12-31 21:05 foo"), 21 * 3600 + 5 * 60)
+  expect_equal(excel_time_to_numeric("1899-12-31 0:05 foo"), 5 * 60)
+  expect_equal(excel_time_to_numeric("1899-12-31 00:05 foo"), 5 * 60)
+  expect_equal(excel_time_to_numeric("1899-12-31 21:05:20 foo"), 21 * 3600 + 5 * 60 + 20)
+})
+
+test_that("excel_time_to_numeric, POSIX times treat no time as midnight but only if there is a space indicating a mostly-well-formed date-time object.", {
```

Comment:
I feel like there must be some story behind this use case 😆

Reply:
You'd be amazed at the trash that I have to sort through in some of my source data.  (Or maybe as the originator of `janitor`, you wouldn't 😄 .)  I sometimes get fields that have multiple data points concatenated together.  Dates and times are especially bad.  As long as it fully matches an expected format (maybe with extra bits at the end), I wanted to let it in.